Jenkinsfile

```diff
@@ -6,116 +6,115 @@ pipeline {
-                timeout(time: 30, unit: 'MINUTES')
-            }
-            environment {
-                JENKINS_REPO = "${infra.isTrusted() ? 'jenkins' : 'jenkins4eval'}/inbound-agent"
-            }
-            steps {
-                script {
-                    def branchName = "${env.BRANCH_NAME}"
-                    infra.withDockerCredentials {
-                        if (branchName ==~ 'master') {
-                            // publish the images to Dockerhub
-                            sh '''
-                            docker buildx create --use
-                            docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
-                            docker buildx bake --push --file docker-bake.hcl linux
-                            '''
-                        } else if (env.TAG_NAME == null) {
-                            sh 'make build'
-                            sh 'make test'
-                            sh '''
-                                docker buildx create --use
-                                docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
-                                docker buildx bake --file docker-bake.hcl linux
-                            '''
-                        }
+        stage('Build') {
+            parallel {
+                stage('Windows') {
+                    options {
+                        timeout(time: 60, unit: 'MINUTES')
                     }
-
-                    if(env.TAG_NAME != null) {
-                        def tagItems = env.TAG_NAME.split('-')
-                        if(tagItems.length == 2) {
-                            def remotingVersion = tagItems[0]
-                            def buildNumber = tagItems[1]
-                            // we need to build and publish the tag version
-                            infra.withDockerCredentials {
-                                sh """
-                                docker buildx create --use
-                                docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
-                                export REMOTING_VERSION=$remotingVersion
-                                export BUILD_NUMBER=$buildNumber
-                                export ON_TAG=true
-                                docker buildx bake --push --file docker-bake.hcl linux
-                                """
+                    environment {
+                        DOCKERHUB_ORGANISATION = "${infra.isTrusted() ? 'jenkins' : 'jenkins4eval'}"
+                    }
+                    stages {
+                        stage('Build and Test') {
+                            // This stage is the "CI" and should be run on all code changes triggered by a code change
+                            when {
+                                not { buildingTag() }
+                            }
+                            steps {
+                                script {
```

Comment:
visualisation is broken in blue ocean unfortunately as it can't handle this case 😢

Reply:
As far as I remember, it is the main reason why nesting matrix/parallel are not supported in declarative: to avoid breaking the UI.

But in fact, it looks better once the build is finished (wether it's successful, failing or aborted):

- Check https://ci.jenkins.io/job/Packaging/job/docker-inbound-agent/job/PR-280/12/pipeline-graph/
- Compare in BlueOcean the following screenshot:

  - "Broken" graph while it's building (I use quotes because it is still usable)
<img width="1133" alt="Capture d’écran 2022-09-29 à 06 38 41" src="https://user-images.githubusercontent.com/1522731/193010578-766d986b-6a0b-43e3-9e5c-ece34003b061.png">

  - Fixed graph once built:

<img width="741" alt="Capture d’écran 2022-09-29 à 06 38 28" src="https://user-images.githubusercontent.com/1522731/193010754-af3f8a48-2785-4aa2-bb7b-81db721dbb1b.png">


=> when using a clean agent (with no priori cache), the new build jumped from ~39min to ~32 min

Reply:
interesting that it works after complete

Reply:
> interesting that it works after complete

Yep, I realized it while looking to this comment: I was sure it was working and just discovered that's it is broken "while" building.